sync-core/src/test/java/com/cloudant/sync/query/IndexTest.java

```diff
@@ -0,0 +1,135 @@
+        assertThat(index.indexType, is("text"));
+        assertThat(index.indexSettings.size(), is(1));
+        assertThat(index.indexSettings.get("tokenize"), is("simple"));
+    }
+
+    @Test
+    public void returnsNullWhenNoFields() {
+        Index index = Index.setUp(null, indexName);
+        assertThat(index, is(nullValue()));
+
+        index = Index.setUp(new ArrayList<Object>(), indexName);
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenNoIndexName() {
+        Index index = Index.setUp(fieldNames, null);
+        assertThat(index, is(nullValue()));
+
+        index = Index.setUp(fieldNames, "");
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenNoIndexType() {
+        Index index = Index.setUp(fieldNames, indexName, null);
+        assertThat(index, is(nullValue()));
+
+        index = Index.setUp(fieldNames, indexName, "");
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenInvalidIndexType() {
+        Index index = Index.setUp(fieldNames, indexName, "blah");
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void correctlyIgnoresIndexSettings() {
+        Map<String, String> indexSettings = new HashMap<String, String>();
+        indexSettings.put("tokenize", "porter");
+        // json indexes do not support index settings.  Index settings will be ignored.
+        Index index = Index.setUp(fieldNames, indexName, "json", indexSettings);
+        assertThat(index.indexSettings, is(nullValue()));
+    }
+
+    @Test
+    public void correctlySetsIndexSettings() {
+        Map<String, String> indexSettings = new HashMap<String, String>();
+        indexSettings.put("tokenize", "porter");
+        // text indexes support the tokenize setting.
+        Index index = Index.setUp(fieldNames, indexName, "text", indexSettings);
+        assertThat(index.indexSettings.size(), is(1));
+        assertThat(index.indexSettings.get("tokenize"), is("porter"));
+    }
+
+    @Test
+    public void comparesIndexTypeAndReturnsInEquality() {
+        Index index = Index.setUp(fieldNames, indexName);
+        assertThat(index.compareIndexTypeTo("text", null), is(false));
+    }
+
+    @Test
+    public void comparesIndexTypeAndReturnsEquality() {
+        Index index = Index.setUp(fieldNames, indexName);
+        assertThat(index.compareIndexTypeTo("json", null), is(true));
+    }
+
+    @Test
+    public void comparesIndexSettingsAndReturnsInEquality() {
+        Index index = Index.setUp(fieldNames, indexName, "text");
+        assertThat(index.compareIndexTypeTo("text", "{\"tokenize\":\"porter\"}"), is(false));
+    }
+
+    @Test
+    public void comparesIndexSettingsAndReturnsEquality() {
+        Index index = Index.setUp(fieldNames, indexName, "text");
+        assertThat(index.compareIndexTypeTo("text", "{\"tokenize\":\"simple\"}"), is(true));
+    }
```

Comment:
Adding a test for this is not needed presently since an Index instance will only initialize if valid parms are provided.  Currently this is limited to "tokenize".  So we can only have one entry in indexSettings.  Once we add support for more valid parms I will add more tests to ensure this scenario is tested.  I have however, added in logic to ensure that this problem is not encountered in the future.  See comment above.